sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTranslation.java

```diff
@@ -455,8 +454,8 @@ public Row toConfigRow(Write<?> transform) {
         fieldValues.put(
             "json_time_partitioning", toByteArray(transform.getJsonTimePartitioning().get()));
       }
-      if (transform.getClustering() != null) {
-        fieldValues.put("clustering", toByteArray(transform.getClustering()));
+      if (transform.getJsonClustering() != null) {
```

Comment:
Will method name change breaking upgrade compatibility? CC: @chamikaramj

Reply:
Yeah, it will since old Beam versions will not have this field.

We have to fork when using this field to maintain compatibility.

Please see the change to BigQueryIOTranslation below for an example.

https://github.com/apache/beam/commit/32f38b92c3652058b0191b94105e0b9441ab965e

Reply:
Ended up using another workaround so we can keep using the same field name and type

Reply:
Actually it turns out translation with clustering has always been broken. The `Clustering` object is not serializable, so this line would always throw an error:
https://github.com/apache/beam/blob/6366bd4c8901063467865277891ffc22769db1b7/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTranslation.java#L478

We don't have a test for this so we didn't notice. 

Since it has always been broken, I think there's no harm to use a different field and there's no need to fork. wdyt @chamikaramj ?

Reply:
You don't need to fork here but you need to fork at `fromConfigRow` if you introduce a new field since trying to pull a non existent field from a Row will result in an error.

Reply:
Thanks, just added a fork